### PR TITLE
Fixes #394 and #395 (ta and  obio dual ocean)

### LINKED ISSUE #394: Bug in GEOS_SurfaceGridComp.F90: TA used in Run2 without MAPL_GetPointer
It looks like @yvikhlya found a bug in GEOS_SurfaceGridComp.F90. Namely, there is this code in the Run2 procedure:

https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/eb7e1994a5169a285b10a93df3cecaad632fa7c5/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOS_SurfaceGridComp.F90#L5767-L5820

and specifically:

https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/eb7e1994a5169a285b10a93df3cecaad632fa7c5/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOS_SurfaceGridComp.F90#L5808-L5819

The fun part is, no where in Run2 is TA "gotten" as it were by `MAPL_GetPointer`. It's declared:

https://github.com/GEOS-ESM/GEOSgcm_GridComp/blob/eb7e1994a5169a285b10a93df3cecaad632fa7c5/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/GEOS_SurfaceGridComp.F90#L4896

and then the next mention of it is line 5808.

The change came in with https://github.com/GEOS-ESM/GEOSgcm_GridComp/commit/4e3110e097500a64bac0069cc4ba82919c8fe04d with the commit message: "Merged with S2S-3 cvs tag". So I'm wondering if that CVS tag *does* have a `MAPL_GetPointer` that was never transferred to Git? Can @yvikhlya or @sanAkel take a look at that?

Also: I guess I'll ping/tag @gmao-rreichle as his name is around this code. And maybe @atrayano because, well, I get confused when we get near tiles.

### PR DESCRIPTION
**Pull request by atrayano**
This version fixes two issues: #394 and #395. The first of them was fixes by @yvikhlya in this branch:
 https://github.com/GEOS-ESM/GEOSgcm_GridComp/tree/bugfix/yury/dual_ocean
and my fix for #395 went on top

### CODE REVIEW

**Comment by yvikhlya** [comment]
Let me test this real quick.

File: GEOSogcm_GridComp/GEOS_OgcmGridComp.F90 [hunk 1]
```diff
@@ -235,6 +235,7 @@ subroutine SetServices ( GC, RC )
        ORAD = MAPL_AddChild(GC, NAME='ORAD', SS=OradBioSetServices, RC=STATUS)
        VERIFY_(STATUS)
     else
+       OBIO = 0
```
Comment (sanAkel) [inline GEOSogcm_GridComp/GEOS_OgcmGridComp.F90:238]: @atrayano a comment would help, which would make it clear that (if I understand correctly!) there is no OBIO child

**Comment by atrayano** [comment]
This would be great! Thank you, Yuri

My “testing” was to make sure the code compiled 😊

File: GEOSogcm_GridComp/GEOS_OgcmGridComp.F90 [hunk 2]
```diff
@@ -2234,6 +2236,7 @@ subroutine RUN ( GC, IMPORT, EXPORT, CLOCK, RC )
           CHLD = (/OBIO,ORAD,SEAICEd,OCEAN/)
           DO N=1, size(CHLD)
              ID = CHLD(N)
+             if (ID <= 0) cycle
```
Comment (sanAkel) [inline GEOSogcm_GridComp/GEOS_OgcmGridComp.F90:2239]: Same comment as above, please add a single description. Thank you!

Comment (atrayano) [reply GEOSogcm_GridComp/GEOS_OgcmGridComp.F90:2239]: @sanAkel I agree, but my main motivation was to give @yvikhlya something quick to test.

**Comment by yvikhlya** [comment]
Both these fixes seem to work, but debug build still crashes with floating point overflow error and no detailed traceback for some other reason. Regular build runs, but according to print out, memory usage grows all the time.

**Comment by sanAkel** [comment]
> Both these fixes seem to work, but debug build still crashes with floating point overflow error and no detailed traceback for some other reason. Regular build runs, but according to print out, memory usage grows all the time.

@yvikhlya valgrid?

**Comment by sanAkel** [comment]
> Memory leak issue is solved on a different branch. Debug build crash should probably be another issue too. I think we should merge these updates into develop, because these are necessary bug fixes.

I am opposed to merging code which is known to be buggy and crashes. Particularly if the reason(s) for crash are not know. cc: @tclune, @mathomp4, @sdrabenh, @wmputman

**Comment by yvikhlya** [comment]
Memory leak issue is solved on a different branch. Debug build crash should probably be another issue too. I think we should merge these updates into develop, because these are necessary bug fixes.


> > Memory leak issue is solved on a different branch. Debug build crash should probably be another issue too. I think we should merge these updates into develop, because these are necessary bug fixes.
> 
> I am opposed to merging code which is known to be buggy and crashes. Particularly if the reason(s) for crash are not know. cc: @tclune, @mathomp4, @sdrabenh, @wmputman

Proposed fixes are correct and are **necessary** to run the model in dual ocean mode. The crash which we experience in debug build is a different issue and I would suggest to resolve it in different PR.

**Comment by sanAkel** [comment]
> > > Memory leak issue is solved on a different branch. Debug build crash should probably be another issue too. I think we should merge these updates into develop, because these are necessary bug fixes.
> > 
> > 
> > I am opposed to merging code which is known to be buggy and crashes. Particularly if the reason(s) for crash are not know. cc: @tclune, @mathomp4, @sdrabenh, @wmputman
> 
> Proposed fixes are correct and are **necessary** to run the model in dual ocean mode. The crash which we experience in debug build is a different issue and I would suggest to resolve it in different PR.

Is that
>The crash which we experience in debug build is a different issue

**different issue** noted some where, number?

**Pull request merged**